<?xml version="1.0" ?>
<launch>
    <node name="youbot_action_server" pkg="youbot_moveit" type="action_server.py"/>
    <node name="base_controller" pkg="youbot_moveit" type="base_controller.py" />
<!--    <include file="$(find youbot_moveit)/launch/youbot_moveit.launch"/>
    <node name="base_controller_test" pkg="youbot_moveit" type="first_controller.py"/>-->

    <!-- <group ns="moveit"> -->
      <!-- By default, we do not start a database (it can be large) -->
      <arg name="db" default="false" />
      <!-- Allow user to specify database location -->
      <arg name="db_path" default="$(find youbot_moveit)/default_warehouse_mongo_db" />

      <!-- By default, we are not in debug mode -->
      <arg name="debug" default="false" />
      <arg name="pipeline" default="ompl" />

      <!-- Load the URDF, SRDF and other .yaml configuration files on the param server -->
      <include file="$(find youbot_moveit)/launch/planning_context.launch">
        <arg name="load_robot_description" value="true"/>
      </include>


      <!-- Run the main MoveIt executable without trajectory execution (we do not have controllers configured by default) -->
      <include file="$(find youbot_moveit)/launch/move_group.launch">
        <arg name="allow_trajectory_execution" value="true"/>
        <arg name="fake_execution" value="false"/>
        <arg name="info" value="true"/>
        <arg name="debug" value="$(arg debug)"/>
        <arg name="pipeline" value="$(arg pipeline)"  />
      </include>

      <!-- Run Rviz and load the default config to see the state of the move_group node -->
      <include file="$(find youbot_moveit)/launch/moveit_rviz.launch">
        <arg name="config" value="true"/>
        <arg name="debug" value="$(arg debug)"/>
      </include>

      <!-- If database loading was enabled, start mongodb as well -->
      <include file="$(find youbot_moveit)/launch/default_warehouse_db.launch" if="$(arg db)">
        <arg name="moveit_warehouse_database_path" value="$(arg db_path)"/>
      </include>

      <node name="moveit_robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" respawn="true" output="screen" >
      <param name="tf_prefix" value="moveit" />
      </node>
      <!-- If needed, broadcast static tf for robot root -->
      <node pkg="tf" type="static_transform_publisher" name="virtual_joint_broadcaster_0" args="0 0 0 0 0 0 /odom /moveit/odom 100" />

      <!-- <node pkg="tf2_ros" type="static_transform_publisher" name="virtual_joint_broadcaster_1" args="0 0 0 0 0 0 base_footprint odom " /> -->


      <!--create vitrual joint state publisher for moveit, we know that Gazebo or real robot only publish 5 DOF for their joint_state, but now we need 8 DOF joint_state -->
      <node name="moveit_joint_state_publisher" pkg="youbot_moveit" type="m_joint_state.py" />

      
    <!-- </group> -->
    

    <!-- relay the moviet joint_states message with gazebo joint_states message-->
    <node pkg="topic_tools" type="relay" name="relay2" args="/joint_states /moveit/joint_states " /> 
</launch>
